gpii/node_modules/flowManager/src/userKeyaction.js

```diff
@@ -0,0 +1,221 @@
+/*
+ * GPII User Login Component
+ *
+ * Copyright 2012 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013)
+ * under grant agreement no. 289016.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+(function () {
+
+    "use strict";
+
+    var fluid = require("infusion");
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.request.flowManager.userKeyactions");
+
+
+
+    gpii.request.flowManager.userKeyactions.handleResetToken = function (activeToken, that) {
+        if (activeToken === undefined) { // if no user is logged in
+            that.errorResponse("No users currently logged in - nothing to reset");
+        } else {
+            fluid.log("Logging out the user with token " + activeToken);
+            that.logoutUser(activeToken);
+        }
+    };
+
+    // keyactivity/<mytoken>:
+    // * If no user is logged in, user <mytoken> will be logged in
+    // * If <mytoken> is logged in, <mytoken> will be logged out
+    // * If another user is logged in, nothing happens
+
+    // If "reset" is used as the <mytoken> for any of the above URLs, any currently logged in user will be logged out
+    gpii.request.flowManager.userKeyactions.keyAction = function (userToken, lifecycleManager, that) {
```

Comment:
Inconsistent capitalisation here - what is a "key action"? Have we explained in our docs or community communications what one is?